client/js/widgets/widget.js

```diff
@@ -1281,6 +1282,52 @@ export class Widget extends StateManaged {
         }
       }
 
+      if(a.func == 'SCORE') {
+        setDefaults(a, { mode: 'set', property: 'score', seats: null, round: null, value: null});
+        if([ 'set', 'inc', 'dec' ].indexOf(a.mode) == -1) {
+          problems.push(`Warning: Mode ${a.mode} interpreted as set.`);
+          a.mode = 'set'
+        }
+        
+        if(a.value === null)
+          a.value = a.mode=='set' ? 0 : 1;
+        if(isNaN(parseFloat(a.value))) {
+          problems.push(`value ${a.value} must be a number, assuming 0.`);
+          a.value = 0;
+        }
+        a.value = parseFloat(a.value);
+
+        let round = a.round;
+        if(round != null && (isNaN(parseInt(round)) || round < 1)) {
+          problems.push(`round ${a.round} must be null or a positive integer, assuming null.`);
+          round = null;
+        }
+
+        const seats = widgetFilter(w => w.get('type')=='seat' && (a.seats===null || asArray(a.seats).includes(w.get('id'))));
+
+        const relation = (a.mode == 'set') ? '=' : (a.mode == 'dec' ? '-' : '+');
+        for(let i=0; i < seats.length; i++) {
+          let newScore = [...asArray(seats[i].get(a.property) || 0)];
```

Comment:
Doesn't `asArray` always return an array? In that case `[...a]` shouldn't change anything, should it?

Reply:
`asArray` returns the original array if it is already an array. Here I need a deep copy, so I think I need `...`.

Reply:
The property being retrieved at the beginning of the loop is being set at the end of the loop, so I don't see how it matters that you are working with a copy rather than a reference.

Reply:
It does matter, because if you just use the original array, the updates to array elements actually change the property value without generating a delta. Then the `set` at the end doesn't cause a delta to be generated either. Happy to listen to other suggestions.